src/Model/Contact.php

```diff
@@ -98,6 +98,26 @@ class Contact extends BaseObject
 	 * @}
 	 */
 
+	/**
+	 * @brief Get the basepath for a given contact link
+	 *
+	 * @param string $url The contact link
+	 *
+	 * @return string basepath
+	 */
+	public static function getBasepath($url)
+	{
+		$data = Probe::uri($url);
+		if (!empty($data['baseurl'])) {
+			return $data['baseurl'];
+		}
+
+		$urlarr = explode('/profile/', $url);
```

Comment:
Isn't this function limited to DFRN contacts?

Reply:
Yeah.